extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -51,19 +43,36 @@ const UpgradeNudge = ( { planName, planPathSlug, postId, postType } ) => (
 	</Warning>
 );
 
-export default withSelect( ( select, { plan: planSlug } ) => {
-	const plan = select( 'wordpress-com/plans' ).getPlan( planSlug );
+export default compose( [
+	withSelect( ( select, { plan: planSlug } ) => {
+		const plan = select( 'wordpress-com/plans' ).getPlan( planSlug );
 
-	// WP.com plan objects have a dedicated `path_slug` field, Jetpack plan objects don't
-	// For Jetpack, we thus use the plan slug with the 'jetpack_' prefix removed.
-	const planPathSlug = startsWith( planSlug, 'jetpack_' )
-		? planSlug.substr( 'jetpack_'.length )
-		: get( plan, [ 'path_slug' ] );
+		// WP.com plan objects have a dedicated `path_slug` field, Jetpack plan objects don't
+		// For Jetpack, we thus use the plan slug with the 'jetpack_' prefix removed.
+		const planPathSlug = startsWith( planSlug, 'jetpack_' )
+			? planSlug.substr( 'jetpack_'.length )
+			: get( plan, [ 'path_slug' ] );
 
-	return {
-		planName: get( plan, [ 'product_name_short' ] ),
-		planPathSlug,
-		postId: select( 'core/editor' ).getCurrentPostId(),
-		postType: select( 'core/editor' ).getCurrentPostType(),
-	};
-} )( UpgradeNudge );
+		const postId = select( 'core/editor' ).getCurrentPostId();
+		const postType = select( 'core/editor' ).getCurrentPostType();
+
+		const upgradeUrl = addQueryArgs(
+			`https://wordpress.com/checkout/${ getSiteFragment() }/${ planPathSlug }`,
+			{
+				redirect_to: `/${ postType }/${ getSiteFragment() }/${ postId }`,
```

Comment:
This will stay in Calypso, right? I wonder about CPTs 😬 

Do we need to check that it's a valid route? `page | post` and possibly `testimonial` | `portfolio` CPTs as well?

Reply:
I do believe Calypso supports _any_ post type that Jetpack site happens to support, so leaving it open might be best.

`wp_block` post type might be exception tho, I don't think Calypso Gutenberg iframe supports that. (Might've changed since I last checked)

Reply:
> This will stay in Calypso, right? I wonder about CPTs 
> 
> Do we need to check that it's a valid route? `page | post` and possibly `testimonial` | `portfolio` CPTs as well?

Yeah, good point. I admit that I didn't think it all through: It's true that Calypso's routes for `testimonial` and `portfolio` are `/types/jetpack-testimonial/` and `/types/jetpack-portfolio/`, respectively. I _think_ we have an endpoint providing this sort of information (but using it here would significantly increase complexity of this PR, so I was thinking of this as a v1).

OTOH, I'm not sure we even offer our blocks on anything else than posts and pages?

Reply:
Okay, I just checked: I can edit portfolio items under `https://wordpress.com/block-editor/edit/jetpack-portfolio/:site/:postId`, _and_ insert the Simple Payments block there, so the redirect link generated here is indeed incorrect and should be fixed.

For a v1, we could add a mapping table, i.e. something like

```js
const cpt = {
portfolio: 'types/jetpack-portfolio',
testimonial: 'types/jetpack-testimonial'
```

(or something similar) and use that to generate the redirect link.

Reply:
Alternatively, this is the endpoint that has the relevant information:

https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/post-types/

Probably not too hard to add a Gutenberg-style store for that :thinking:

Reply:
Is there a query param we can add to force the block editor when the iframe loads in Calypso? @Automattic/serenity 

I just hit an edge case where I'm in the block editor via WP Admin, purchase via nudge, then land in the iframe with the classic editor activated and a modal to use the new editor.

Reply:
> I do believe Calypso supports _any_ post type that Jetpack site happens to support, so leaving it open might be best.

The problem I'm considering is something like:

Editing WooCommerce Product -> click nudge -> pay -> https://wordpress.com/product/SITE_SLUG/POST_ID

That leaves us at a dead end.

Reply:
>I just hit an edge case where I'm in the block editor via WP Admin, purchase via nudge, then land in the iframe with the classic editor activated and a modal to use the new editor.

That doesn't seem right and likely a bug? 🤔

Reply:
>That doesn't seem right and likely a bug?

Related paYKcK-64-p2

Reply:
> That doesn't seem right and likely a bug? 🤔

In my case I had just switched to the classic editor for some testing. I think continuing in the classic editor when it's been selected is expected behavior.

Reply:
To cycle back to the CPT route question, seems like Calypso is fine with simply prepending `/types/` in pretty much all cases (i.e. except for `post` and `page`):

https://github.com/Automattic/wp-calypso/blob/d0383bd0ccc3f60dea7e9aa559beaae0df80bca2/client/my-sites/sidebar/site-menu.jsx#L266

Reply:
Going to add a commit to take care of CPTs.